src/Microsoft.Tye.Hosting/DockerRunner.cs

```diff
@@ -260,21 +270,27 @@ private async Task StopContainerAsync(Tye.Hosting.Model.Service service)
             }
         }
 
-        private static async Task<bool> DetectDockerInstalled()
+        private static string? GetUserSecretsPathFromSecrets()
         {
-            // Detect Docker installation
-            try
-            {
-                await ProcessUtil.RunAsync("docker", "version", throwOnError: false);
-                return true;
-            }
-            catch (Exception)
+            const string userSecretsFallbackDir = "DOTNET_USER_SECRETS_FALLBACK_DIR";
+
+            // For backwards compat, this checks env vars first before using Env.GetFolderPath
+            var appData = Environment.GetEnvironmentVariable("APPDATA");
+            var root = appData                                                                   // On Windows it goes to %APPDATA%\Microsoft\UserSecrets\
+                       ?? Environment.GetEnvironmentVariable("HOME")                             // On Mac/Linux it goes to ~/.microsoft/usersecrets/
+                       ?? Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)
+                       ?? Environment.GetFolderPath(Environment.SpecialFolder.UserProfile)
+                       ?? Environment.GetEnvironmentVariable(userSecretsFallbackDir);            // this fallback is an escape hatch if everything else fails
+
+            if (string.IsNullOrEmpty(root))
             {
-                // Unfortunately, process throws
-                return false;
+                return null;
             }
-        }
 
+            return !string.IsNullOrEmpty(appData)
```

Comment:
Should we be using OSPlatform instead of checking for AppData?

Reply:
I I just copied this logic: https://github.com/dotnet/extensions/blob/64140f90157fec1bfd8aeafdffe8f30308ccdf41/src/Configuration/Config.UserSecrets/src/PathHelper.cs#L27-L62